tests/utils/misc/index.js

```diff
@@ -0,0 +1,117 @@
+const path = require('path');
+const fse = require('fs-extra');
+const execSync = require('child_process').execSync;
+const BbPromise = require('bluebird');
+const replaceTextInFile = require('../fs').replaceTextInFile;
+
+const logger = console;
+
+const testRegion = 'us-east-1';
+
+const testServiceIdentifier = 'integ-test';
+
+const serverlessExec = path.join(__dirname, '..', '..', '..', 'bin', 'serverless');
+
+const serviceNameRegex = new RegExp(`${testServiceIdentifier}-d+`);
+
+function getServiceName() {
+  const hrtime = process.hrtime();
+  return `${testServiceIdentifier}-${hrtime[1]}`;
+}
+
+function deployService() {
+  execSync(`${serverlessExec} deploy`, { stdio: 'inherit' });
+}
+
+function removeService() {
+  execSync(`${serverlessExec} remove`, { stdio: 'inherit' });
+}
+
+function replaceEnv(values) {
+  const originals = {};
+  for (const key of Object.keys(values)) {
+    if (process.env[key]) {
+      originals[key] = process.env[key];
+    } else {
+      originals[key] = 'undefined';
+    }
+    if (values[key] === 'undefined') {
+      delete process.env[key];
+    } else {
+      process.env[key] = values[key];
+    }
+  }
+  return originals;
+}
+
+function createTestService(templateName, tmpDir, testServiceDir) {
+  const serviceName = getServiceName();
+
+  fse.mkdirsSync(tmpDir);
+  process.chdir(tmpDir);
+
+  // create a new Serverless service
+  execSync(`${serverlessExec} create --template ${templateName}`, { stdio: 'inherit' });
```

Comment:
Also, maybe it would be better to have pre-prepared service setups? Instead of generating them with `create`.

Thing is that this way we will separate testing of `create` output from testing of other features.

e.g. I imagine that at some point we may update `create` output with some changes, while users's existing services won't receive those, our integration tests will automatically be updated with, while I think for reliability they shouldn't.

Reply:
Interesting. Definitely smth. we should discuss.

Right now we're copying our test template over after creating the service. Not entirely sure if I'm following the argument. The main idea is to have a complete E2E service from a clean slate / new users perspective.

Reply:
With above what we're testing is:

_Will given change affect how new service (created from scratch via `sls create`) prerforms_

While I think we should rather test:

_Will given change affect a service that was already setup with previous version of SLS._ 

(so: whether upgrading SLS on those services will break them or not)

Difference is of course subtle, but still I think test will probably be more bulletproof if mock services are not affected by eventual changes we apply to `sls create` outcome.